test/strscan/test_stringscanner.rb

```diff
@@ -783,6 +783,14 @@ def test_scan_aref_repeatedly
     assert_equal "t",      s[1]
     assert_equal "ring",   s[2]
   end
+  
+  def test_named_captures
+    s = "foobarbaz"
+    re = /(?<f>foo)(?<r>bar)(?<z>baz)/
+    scan = StringScanner.new(s)
+    scan.match? re
```

Comment:
We can simplify this:

```suggestion
    scan = StringScanner.new("foobarbaz"/)
    scan.match?(/(?<f>foo)(?<r>bar)(?<z>baz))
```

Reply:
This is much better. Thank you.